code/modules/projectiles/pins.dm

```diff
@@ -23,10 +23,14 @@
 	if(proximity_flag)
 		if(istype(target, /obj/item/gun))
 			var/obj/item/gun/G = target
-			if(G.pin && (force_replace || G.pin.pin_removeable))
-				G.pin.forceMove(get_turf(G))
```

Comment:
why is G in nullspace instead of in the mob's contents?

Reply:
mob's contents are considered as being in nullspace when using get_turf. this is an ancient issue.

Reply:
you sure? couldn't reproduce in a test project

Reply:
huh.... yeah seems that's not an issue anymore i guess?

either way, drop_location fixes the pin being force_moved onto the top turf instead of ending up inside of a locker if someone's inside of it or something.